src/OpenDiffix.Core/CommonTypes.fs

```diff
@@ -349,6 +350,20 @@ module ExecutionContext =
   let makeDefault () =
     fromQueryContext (QueryContext.makeDefault ())
 
+module Bucket =
+  let make group aggregators executionContext =
+    {
+      Group = group
+      Aggregators = aggregators
+      ExecutionContext = executionContext
+      Attributes = Dictionary<string, Value>()
+    }
+
+  let getAttribute attr bucket =
+    bucket.Attributes |> Dictionary.getOrDefault attr Null
```

Comment:
Hmm, I don't think we need `Null` as an attr. If you want a flag you can put a `Boolean`.

Reply:
But anyhow, it's just a convenience function. We can have `tryGet` for the `option` version.

Reply:
I was thinking in case we want to store a nullable computation, like the result of a `sum` aggregation.
With this interface, the caller can't differentiate between the attribute doesn't apply case and it does, but it is null case.
But let's cross that bridge when we get there.